app/V1Module/security/Policies/UploadedFilePermissionPolicy.php

```diff
@@ -58,6 +58,30 @@ public function isAuthorOfSupplementaryFileExercises(Identity $identity, Uploade
     return FALSE;
   }
 
+
+  public function isInGroupUserSupervises(Identity $identity, UploadedFile $file)
```

Comment:
I understand that the name is long enough as is, but it should probably reflect the fact that this assertion only works for supplementary (test) files.

Reply:
Any suggestions?

Reply:
isSupplementaryInGroupUserSupervises? That's awfully long. Is it necessary for the method to only work on supplementary files? Don't additional files have a list of exercises too?